src/App.jsx

```diff
@@ -1,248 +1,303 @@
+}
+
+const owners = categoriesFromServer.map(own => ({
+  user: getUserById(own.ownerId)
+}));
+
+function getOwnUsersById(userId) {
+  return owners.filter(ownUser => ownUser.user.id === userId)
+    || null;
+}
+
+const products = productsFromServer.map(product => ({
+  ...product,
+  category: getCategoryById(product.categoryId),
+  user: getOwnUsersById(product.userId)
+}));
+
+
+function getFilterProducts(products, query) {
+  let productFilter = [...products];
+
+  if (query) {
+    const normalizeQuery = query.toLowerCase().trim();
+
+    productFilter = productFilter.filter(product => (
+      product.name.toLowerCase().startsWith(normalizeQuery)
+    ));
+  }
+
+  return productFilter;
+}
+
+console.log(products)
+
+export const App = () => {
+  // const category = products.map(category => category.name)
+  // console.log(category)
+  // const [sorted, setSorted] = useState(category)
+  const [select, setSelected] = useState('')
+  const [query, setQuery] = useState('');
+
+  const productForRendering = getFilterProducts(products, query);
+
+  const sortByGrocery = () => {
+    setSelected('Grocery')
+  }
+
+  const sortByDrinks = () => {
+    setSelected('Drinks')
+  }
+
+  const sortByFruits = () => {
+    setSelected('Fruits')
+  }
+
+  const sortByElectronics = () => {
+    setSelected('Electronics')
+  }
+
+  const sortByClothes = () => {
+    setSelected('Clothes')
+  }
+
+
+  const deleteBtn = () => {
+    setQuery('')
+  }
 
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
+  const sortByAll = () => {
+    setSelected('All')
+  }
 
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
+  const sortByRoma = () => {
+    setSelected('Roma')
+  }
 
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+  const sortByAnna = () => {
+    setSelected('Anna')
+  }
+
+  const sortByMax = () => {
+    setSelected('Max')
+  }
+
+  const sortByJohn = () => {
+    setSelected('John')
+  }
+
+  return (
+    <div className="section">
+      <div className="container">
+        <h1 className="title">Product Categories</h1>
+
+        <div className="block">
+          <nav className="panel">
+            <p className="panel-heading">Filters</p>
+
+            <p className="panel-tabs has-text-weight-bold">
+              <a
+                onClick={sortByAll}
+                className={select === 'All' ? "is-active" : ""}
+                data-cy="FilterAllUsers"
+                href="#/"
+              >
+                All
+              </a>
+
+              <a
+                onClick={sortByRoma}
+                className={select === 'Roma' ? "is-active" : ""}
+                data-cy="FilterUser"
+                href="#/"
+              >
+                Roma
+              </a>
```

Comment:
You need to use `map` method for users from server here instead of repeating each user